digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -184,9 +184,13 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
       }
       if (GeometryUtils.areAdjacent(newLink.geometry.head, endP) ||
         GeometryUtils.areAdjacent(newLink.geometry.last, startP))
-        SideCode.TowardsDigitizing
-      else
-        SideCode.AgainstDigitizing
+        if (existingLink.sideCode.equals(AgainstDigitizing))
```

Comment:
This looks suspicious: the adjacency check looks for newLink start being at the end of the previous link or vice versa, but sidecodes don't seem to match.

Reply:
With the code above, it looks like we should be using the existingLink and reverse the direction if needed. A thorough unit test should help with this.

Reply:
I think yes, we should use the existing value and reverse when we need. I'm just not sure if this validation is the most correct one, but I did some tests and it worked. I'll add a unit test